### PR TITLE
ListView drop indicator fixes

### PR DESCRIPTION
**Pull request by reidbarber**
- Fixes roles for root drop indicator
- Adds hidden button for dropping on an item
- Fixes to not render drop indicators if not in drag session
- Fixes insertion indicator z-index to be above row border when rows are selected

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

Test ListView DnD stories for desired behavior/announcements for dropping between, on, and on root.

To test insertion indicator z-index fix:
In 'Reorder' story, select some rows, then drag an item over the selected rows. Insertion indicator should be above the border of the selected rows. 

## 🧢 Your Project:

RSP

### CODE REVIEW

**Comment by snowystinger** [comment]
Couple oddities

in this story https://reactspectrum.blob.core.windows.net/reactspectrum/21b459c582f495a077816bbbd281e5838d1fd7e3/storybook/index.html?path=/story/listview-drag-and-drop--drag-between-lists-root-only
I get no announcement that I'm focused on another list, it just say "tab to navigate to a drop target" but tab doesn't do anything because there's only the one place that I can drop

similar situation with announcements for this story as well https://reactspectrum.blob.core.windows.net/reactspectrum/21b459c582f495a077816bbbd281e5838d1fd7e3/storybook/index.html?path=/story/listview-drag-and-drop--drag-into-folder

also, in the drag into folder story, I can get it to say: you are currently on a button, inside of a blank, if I stay in drag mode long enough on the folder
Screen Shot 2022-05-16 at 4 30 23 PM

**Comment by reidbarber** [comment]
Updated 'drop into' and 'root only' stories to have multiple droppable areas, where announcements are made when navigating between them. The case of not announcing the initial drop target is being handled in a follow-up

File: packages/@react-spectrum/list/src/InsertionIndicator.tsx [hunk 1]
```diff
@@ -20,6 +20,10 @@ export default function InsertionIndicator(props: InsertionIndicatorProps) {
 
   let isDropTarget = dropState.isDropTarget(target);
 
+  if (!isDropTarget && dropIndicatorProps['aria-hidden']) {
+    return null;
+  }
```
Comment (LFDanLu) [inline packages/@react-spectrum/list/src/InsertionIndicator.tsx:25]: Remind me, what was the reason for not rendering the drop indicators until a drag session starts?

File: packages/@react-spectrum/list/src/ListViewItem.tsx [hunk 2]
```diff
@@ -222,6 +225,9 @@ export function ListViewItem<T>(props: ListViewItemProps<T>) {
               }
             </div>
           }
+          {isDropTarget && !dropIndicator?.dropIndicatorProps['aria-hidden'] &&
+            <div role="button" {...visuallyHiddenProps} {...dropIndicator?.dropIndicatorProps} ref={dropIndicatorRef} />
+          }
```
Comment (LFDanLu) [inline packages/@react-spectrum/list/src/ListViewItem.tsx:230]: For other reviewers, this bit is needed to announce the droppable folder/item as a drop indicator. @reidbarber I assume this was added as visually hidden button for a reason? Can we not spread the drop indicator props onto the row itself?

File: packages/@react-spectrum/list/src/RootDropIndicator.tsx [hunk 3]
```diff
@@ -4,22 +4,25 @@ import {useVisuallyHidden} from '@react-aria/visually-hidden';
...
-      aria-selected="false"
-      {...visuallyHiddenProps}
-      {...dropIndicatorProps}
-      ref={dropRef} />
+    <div role="row" aria-hidden={dropIndicatorProps['aria-hidden']}>
+      <div
+        role="gridcell"
+        aria-selected="false"
+        {...visuallyHiddenProps}>
+        <div role="button" {...dropIndicatorProps} ref={ref} />
```
Comment (LFDanLu) [inline packages/@react-spectrum/list/src/RootDropIndicator.tsx:24]: Similar question here, are the `dropIndicatorProps` invalid on the row and thus need to be added to a button?

**Review by LFDanLu — COMMENTED** [review]
Just some questions but otherwise the announcements seem good. As an aside, it may be an idea to have some customized announcements that tell the user to navigate between drop indicators inside a ListView using the arrow keys, just something I noticed when testing in depth. Definitely out of scope of this PR, maybe something to explore as a followup, I'll write an issue

File: packages/@react-spectrum/list/src/InsertionIndicator.tsx (hunk 1 above)
Comment (reidbarber) [reply packages/@react-spectrum/list/src/InsertionIndicator.tsx:25]: I think it makes sense, no reason to have a drop button in the DOM if nothing is being dropped. Some background: https://github.com/adobe/react-spectrum/blob/a0a88da217db1cff67be1bdc94490b3a7002b6f3/packages/%40react-aria/dnd/stories/DroppableGrid.tsx#L329-L331

Comment (reidbarber) [reply packages/@react-spectrum/list/src/InsertionIndicator.tsx:25]: Also https://github.com/adobe/react-spectrum/pull/3055#discussion_r870460265

File: packages/@react-spectrum/list/src/ListViewItem.tsx (hunk 2 above)
Comment (reidbarber) [reply packages/@react-spectrum/list/src/ListViewItem.tsx:231]: I think that works, but this is better. Re: https://github.com/adobe/react-spectrum/pull/3055#discussion_r870484248

File: packages/@react-spectrum/list/src/InsertionIndicator.tsx (hunk 1 above)
Comment (LFDanLu) [reply packages/@react-spectrum/list/src/InsertionIndicator.tsx:25]: I see, thanks. Yeah that makes sense to me, just wanted to make sure there wasn't some weird bug that was also being handled by this change

**Review by snowystinger — APPROVED** [review]
working well in Safari VO MacOS, announcements for dropping on folders is much better

**Review by LFDanLu — APPROVED** [review]
LGTM

**Pull request merged**